packages/host/app/lib/matrix-handlers/index.ts

```diff
@@ -105,16 +102,42 @@ export async function addRoomEvent(context: EventSendingContext, event: Event) {
       ...(resolvedRoom.events ?? []),
       event as unknown as DiscreteMatrixEvent,
     ];
+  }
+}
 
-    let pendingMessage = context.pendingMessages.get(resolvedRoom.roomId);
-    if (
-      pendingMessage &&
-      event.type === 'm.room.message' &&
-      event.content?.msgtype === 'org.boxel.message' &&
-      (event.content as CardMessageContent).clientGeneratedId ===
-        pendingMessage.clientGeneratedId
-    ) {
-      context.pendingMessages.set(resolvedRoom.roomId, undefined);
-    }
+export async function updateRoomEvent(
+  context: EventSendingContext,
+  event: Event,
+  oldEventId: string,
+) {
+  if (event.content?.data && typeof event.content.data === 'string') {
+    event.content.data = JSON.parse(event.content.data);
+  }
+  let { event_id: eventId, room_id: roomId, state_key: stateKey } = event;
+  eventId = eventId ?? stateKey; // room state may not necessary have an event ID
```

Comment:
This is confusing to me - how can event id be two different things?

Reply:
I think this is highlighting the difference between a message event and room state. the event ID is not necessarily guranteed...

Reply:
Thanks @habdelra for the explanation. I didn't add this line, it comes from legacy code.